lib/stan.js

```diff
@@ -343,6 +387,69 @@ Stan.prototype.createConnection = function() {
   });
 };
 
+function processPingResponse(that, msg) {
+  if (msg) {
+    var pingResponse = proto.PingResponse.deserializeBinary(Buffer.from(msg, 'binary').toByteArray());
+    var err = pingResponse.getError();
+    if (err) {
+      closeDueToPing(that, err);
+      return;
+    }
+  }
+  that.pingOut = 0;
+}
+
+function pingServerPeriodically(that) {
+  that.pingTimer = timers.setTimeout(function() {
+    that.pingOut++;
+    if (that.pingOut > that.pingMaxOut) {
+      closeDueToPing(that, new Error(MAX_PINGS_EXCEEDED));
+      return;
+    }
+
+    that.nc.publish(that.pingRequests, that.pingBytes, that.pingInbox);
+
+    pingServerPeriodically(that);
+  }, that.pingInterval);
+}
+
+function cleanupOnClose(that) {
```

Comment:
This should be part of `Stan.close()`, then we can just call `Stan.close()` and insure all the logic of unsubs etc is done in one place.